extension/parquet/parquet-extension.cpp

```diff
@@ -597,7 +597,9 @@ unique_ptr<LocalFunctionData> ParquetWriteInitializeLocal(ExecutionContext &cont
 
 unique_ptr<TableFunctionRef> ParquetScanReplacement(ClientContext &context, const string &table_name,
                                                     ReplacementScanData *data) {
-	if (!StringUtil::EndsWith(StringUtil::Lower(table_name), ".parquet")) {
+	auto lower_name = StringUtil::Lower(table_name);
+	auto path = lower_name.substr(0, lower_name.find('?'));
```

Comment:
I'm wondering here what would happen if we have a parquet file with a ? in its name, I think that will no longer work after this change. I think it makes sense to add a specific case here for the fully specified urls where we check that: the name starts with `s3://`, `https://` or `http://` and the name contains `.parquet?`

Reply:
✔️